Signal/src/Profiles/OWSProfileManager.m

```diff
@@ -264,6 +252,44 @@ - (void)saveUserProfile:(UserProfile *)userProfile
 
 #pragma mark - Local Profile
 
+- (UserProfile *)localUserProfile
+{
+    @synchronized(self)
+    {
+        if (_localUserProfile == nil) {
+            _localUserProfile = [self getOrCreateLocalProfile];
+        }
+
+        return _localUserProfile;
+    }
+}
+
+- (UserProfile *)getOrCreateLocalProfile
```

Comment:
Why bother pulling this out into another method?  Please at least leave a comment that `getOrCreateLocalProfile` should only be called by `localUserProfile`, but IMO better to combine them and avoid the possibility of it being called accidentally.